homeassistant/components/roborock/image.py

```diff
@@ -133,70 +139,54 @@ async def async_image(self) -> bytes | None:
                     translation_key="map_failure",
                 )
             map_data = response[0]
-            self.cached_map = self._create_image(map_data)
+            content = self.create_image(map_data, self.parser)
+            if self.cached_map != content:
+                self.cached_map = content
+                await self.coordinator.map_storage.async_save_map(
+                    self.map_flag,
+                    content,
+                )
```

Comment:
If I understood the code correctly, the map storage will be used only to avoid the first loading of the image.
Can we store the image on shutdown instead of each time we get a new one?
I don't have a Roborock, so my assumptions may be incorrect. I'm the code owner of the Ecovacs integration and the map will be updated during cleaning every few seconds. If this is the case for Roborock, I think it's better to save it less frequently.

Reply:
The image can be fetched from the device at most every `IMAGE_CACHE_INTERVAL` which is 90 seconds, then saved each time.  I think 90 seconds is reasonable compared to every few seconds? But perhaps it's still a good idea.

I also realize this is waiting for the write to disk to finish before displaying to the user, which seems like it will slow things down so this still probably needs adjustment.

Reply:
I've updated this to no block the image serving on the saving for now.

I think we will want to increase the frequency of writing above 90 seconds. For example, i see users report that it needs to keep updating after cleaning is done at least once: https://github.com/home-assistant/core/issues/110588

Given i think 90 seconds is reasonable, how about in a follow up i'll update the storage object to "queue", add a method to flush, then we can toy with increasing the frequency even and update after cleaning is done?